File: problemas/1100/1164.js
const entrada = require('fs')
  .readFileSync('/dev/stdin', 'utf8')
  .split('\n')
  .map(x => Number(x));
const n = entrada.shift();

for (let i = 0; i < n; i++) {
  const x = entrada.shift();
  let soma = 0;

  for (let j = 0; j < x; j++) {
    if (x % j === 0) {
      soma += j;
    }
  }

  console.log(x + (soma === x ? ' eh perfeito' : ' nao eh perfeito'));
}

File: problemas/1100/1164.py
n = int(input())

for i in range(n):
    x = int(input())
    soma = 0

    for j in range(1, x):
        if x % j == 0:
            soma += j

    print(x, 'eh perfeito' if soma == x else 'nao eh perfeito')
